include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -18,89 +18,97 @@
 
 #    include <CL/sycl.hpp>
 
-#    include <iostream>
+#    include <cstddef>
+#    include <exception>
+#    include <functional>
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+#        include <iostream>
+#    endif
 #    include <sstream>
 #    include <stdexcept>
 #    include <vector>
 
 namespace alpaka::experimental
 {
     //! The SYCL device manager.
-    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl>
+    template<typename TSelector>
+    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl<TSelector>>
     {
     public:
         PltfGenericSycl() = delete;
+
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wexit-time-destructors"
+        inline static std::function<void(sycl::exception_list)> async_handler = [](sycl::exception_list exceptions)
+        {
+            auto ss_err = std::stringstream{};
+            ss_err << "Caught asynchronous SYCL exception(s):\n";
+            for(std::exception_ptr e : exceptions)
+            {
+                try
+                {
+                    std::rethrow_exception(e);
+                }
+                catch(sycl::exception const& err)
+                {
+                    ss_err << err.what() << " (" << err.code() << ")\n";
+                }
+            }
+            throw std::runtime_error(ss_err.str());
+        };
+
+        inline static sycl::platform sycl_platform{TSelector {}};
+        inline static std::vector<sycl::device> sycl_devices{sycl_platform.get_devices()};
+        inline static sycl::context sycl_context{sycl_devices, async_handler};
```

Comment:
should this part be excluded from the `#if BOOST_COMP_CLANG` blocks ?

Reply:
No, it is actually the reason why those blocks are there. Otherwise `-Wexit-time-destructors` will be triggered.

Reply:
IIRC the warning will always appear for non-trivial types.

Reply:
I don't understand then: how do you build these objects at all, when `BOOST_COMP_CLANG` is not defined ?

Reply:
Ah, now I see. That was an oversight on my part while doing the refactoring. Thanks for pointing this out.